### PR TITLE
docs: migrate documentation to Antora

### PR DESCRIPTION
**Pull request by Mogztter**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1028,7 +982,6 @@
 				</property>
 			</activation>
 			<modules>
-				<module>neo4j-migrations-test-results</module>
```
Comment (michael-simons) [inline pom.xml:1031]: This is required, otherwise central will fail. see comment above in pom.

File: antora-playbook.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+site:
```
Comment (michael-simons) [inline antora-playbook.yml:1]: Can this be moved into `etc`?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "neo4j-migrations-docs",
+  "version": "1.0.0",
```
Comment (michael-simons) [inline package.json:3]: Can this be derived from the maven version / tag? This is not accurate.

File: pom.xml [hunk 4]
```diff
@@ -68,7 +68,6 @@
 		<module>neo4j-migrations-spring-boot-starter-parent</module>
 		<module>neo4j-migrations-quarkus-parent</module>
 		<module>extensions/neo4j-migrations-annotation-processing</module>
-		<module>extensions/neo4j-migrations-formats-adoc</module>
```
Comment (michael-simons) [inline pom.xml:71]: Why did you remove this module?

**Review by michael-simons — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Mogztter) [reply pom.xml:1031]: I think Intellij removed it, I have no recollection of removing it 🤔

File: pom.xml (hunk 4 above)
Comment (Mogztter) [reply pom.xml:71]: Ditto

File: package.json (hunk 3 above)
Comment (Mogztter) [reply package.json:3]: This package won't be published so the version does not really matter. Though, I need to add `private: true` to make sure that we won't inadvertently publish it: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#private

File: antora-playbook.yml (hunk 2 above)
Comment (Mogztter) [reply antora-playbook.yml:1]: Sure, I can also move the `package.json` and `package-lock.json` to `etc` if you want?

**Comment by michael-simons** [comment]
Sweet. I'll most likely do this in combination with the bump to version 2.

This way I can

- keep the old versioned docs and also maintain the than 1.13.x or 1.14.x branch
- you touch me about Antorra and how to publish it with Neo4j-Labs.

Wdyt?

**Comment by Mogztter** [comment]
Sounds good 👍🏻

File: docs/antora.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+name: migrations
```
Comment (Mogztter) [inline docs/antora.yml:1]: `migrations` is a unique identifier of your "documentation component"

File: docs/antora.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+name: migrations 
+version: ~
```
Comment (Mogztter) [inline docs/antora.yml:2]: `~` means that your documentation component is unversioned. You can define a version at a later point in time, for instance when you will have a maintenance branch for 2.x and a main branch for 3.x.

File: docs/antora.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    url-apidocs: https://michael-simons.github.io/neo4j-migrations/main/site
+    url-projectinfo: https://michael-simons.github.io/neo4j-migrations/main/site
+    url-gh-releases: https://github.com/michael-simons/neo4j-migrations/releases
+    url-release-zip: https://github.com/michael-simons/neo4j-migrations/releases/download/1.12.0/neo4j-migrations-1.12.0.zip
+    groupId: eu.michael-simons.neo4j
+    artifactIdMavenPlugin: neo4j-migrations-maven-plugin
+    artifactIdSpringBoot: neo4j-migrations-spring-boot-starter
+    artifactIdQuarkus: neo4j-migrations-quarkus
+    artifactIdCore: neo4j-migrations
+    branch: main
```
Comment (Mogztter) [inline docs/antora.yml:32]: Theses attributes will be available on all pages (AsciiDoc files) that belongs to this documentation component

File: docs/modules/ROOT/examples/README.adoc [hunk 8]
```diff
@@ -0,0 +1 @@
+../../../../README.adoc
```
Comment (Mogztter) [inline docs/modules/ROOT/examples/README.adoc:1]: We are using symlinks because Antora cannot reference files outside of the documentation component (i.e., outside of the docs directory)

File: docs/antora.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+name: migrations 
+version: ~
+title: Neo4j Migrations Docs
+start_page: ROOT:index.adoc
+nav:
+  - modules/ROOT/nav.adoc
```
Comment (Mogztter) [inline docs/antora.yml:6]: Configure the navigation within your documentation component. Here's an example of the Kafka Integration Documentation (left menu is the navigation):

image

File: docs/antora.yml [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+asciidoc:
+  attributes:
```
Comment (Mogztter) [inline docs/antora.yml:9]: Please note that you cannot (yet) use attribute references. The following *won't* work:

```yaml
    url-gh: https://github.com/michael-simons/neo4j-migrations
    url-gh-releases: {url-gh}/releases
```

File: docs/modules/ROOT/pages/appendix.adoc [hunk 11]
```diff
@@ -41,12 +41,12 @@ We do support the following processing instructions:
 * `<?assert followed by a valid precondition ?>`
 * `<?assume followed by a valid precondition ?>`
 
-Look up valid preconditions <<concepts_preconditions,here>>. The full XMl schema for <<concepts_migrations_catalog-based, catalog-based migrations>> looks like this:
+Look up valid preconditions xref:concepts.adoc#concepts_preconditions[here]. The full XMl schema for xref:concepts.adoc#concepts_migrations_catalog-based[catalog-based migrations] looks like this:
```
Comment (Mogztter) [inline docs/modules/ROOT/pages/appendix.adoc:44]: It's recommended to use the `xref` macro instead of `<<` `>>`.

Since we are using reference to the same module (ROOT), the same documentation component (migrations) and the same version (unversioned) we can use only the file name `concepts.adoc`.

Please note that you can reference pages from other labs components. For instance, you could create a reference to a page of the Kafka Integration Documentation:

```
xref:4.1@kafka:ROOT:kafka-connect.adoc[]
```

https://docs.antora.org/antora/latest/navigation/xrefs-and-link-text/#page-xrefs

File: docs/modules/ROOT/pages/appendix.adoc [hunk 12]
```diff
@@ -41,12 +41,12 @@ We do support the following processing instructions:
...
 
 .migration.xsd
 [source,xml,indent=0,tabsize=2]
 ----
-include::../neo4j-migrations-core/src/main/resources/ac/simons/neo4j/migrations/core/migration.xsd[]
+include::example$migration.xsd[]
```
Comment (Mogztter) [inline docs/modules/ROOT/pages/appendix.adoc:49]: We should avoid using relative paths in Antora. Here we include an "example": https://docs.antora.org/antora/latest/page/include-an-example/

File: docs/modules/ROOT/pages/appendix.adoc [hunk 13]
```diff
@@ -315,7 +315,7 @@ It has no dependencies apart from Neo4j-Migrations itself (neither SDN6 nor Neo4
             </annotationProcessorPath>
         </annotationProcessorPaths>
         <compilerArgs>
-            <arg>-Aorg.neo4j.migrations.catalog_generator.default_catalog_name=V${next-migration-version}__Create_sdn_constraints.xml</arg>
+            <arg>-Aorg.neo4j.migrations.catalog_generator.default_catalog_name=V$\{next-migration-version}__Create_sdn_constraints.xml</arg>
```
Comment (Mogztter) [inline docs/modules/ROOT/pages/appendix.adoc:318]: Escape the `{` otherwise AsciiDoc will treat it as an attribute reference

File: etc/antora/supplemental-ui/partials/header-content.hbs [hunk 14]
```diff
@@ -0,0 +1,33 @@
+<header class="header">
```
Comment (Mogztter) [inline etc/antora/supplemental-ui/partials/header-content.hbs:1]: Override the header of the default UI. We can remove it if we don't plan to host the documentation on GitHub Pages since https://neo4j.com/labs has already a custom site UI.

File: docs/antora.yml (hunk 6 above)
Comment (michael-simons) [reply docs/antora.yml:2]: So, I introduced a 1.x branch for the 1.x stuff I am gonna maintain. In that branch it shall stick with the old way of doing things.

I will continue releasing main (it is still named `main`, but its version is 2.x-SNAPSHOT). What do I need to add here @Mogztter ?

**Comment by michael-simons** [comment]
I have really issues with duplication `migrations.xsd`and a bunch of other files. Right now, documentation is source and source is docs. I will definitely forget updating the copy. Can this be solved different?

**Comment by michael-simons** [comment]
I don't understand why even the top level README.adoc is duplicated.

File: docs/antora.yml (hunk 6 above)
Comment (Mogztter) [reply docs/antora.yml:2]: It depends on how you will manage versions. Antora recommends to create a maintenance branch per minor version.
So, the first version will be "2.0", "2.1", "2.2", "3.0"...

If `main` is 2.x-SNAPSHOT, you can use:

```yaml
version: '2.0-snapshot'
prerelease: true
```

**Comment by Mogztter** [comment]
>I have really issues with duplication migrations.xsdand a bunch of other files. Right now, documentation is source and source is docs. I will definitely forget updating the copy. Can this be solved different?
> I don't understand why even the top level README.adoc is duplicated. How should I maintain it in a sensible fashion?

They are not duplicated, we are using symbolic links.
Antora does not add files that are outside of the documentation component (i.e., outside the `docs` directory) into the content catalog.

> Let’s consider one of the most common uses for the symlink feature in Antora. You have example files you want to include in your documentation, but those files don’t live inside the standard Antora directory structure. In order to make them available to Antora, you need to remap them into the Antora hierarchy.

https://docs.antora.org/antora/latest/symlinks/#remap-files-using-symlinks

**Review by michael-simons — APPROVED** [review]

**Comment by michael-simons** [comment]
@all-contributors please add @Mogztter for code, doc

**Comment by Mogztter** [comment]
> I've put up https://github.com/michael-simons/neo4j-migrations/pull/731 to add @Mogztter! 🎉

Thanks!
Good bot 😄
